src/common/gui/CSurgeSlider.cpp

```diff
@@ -286,17 +289,36 @@ void CSurgeSlider::draw(CDrawContext* dc)
    setDirty(false);
 }
 
-void CSurgeSlider::bounceValue()
+void CSurgeSlider::bounceValue(const bool keeprest)
 {
-   if (value > vmax)
+    if (keeprest) {
+        if (restvalue) {
```

Comment:
Oh I’m sorry to ask for one last change but: we really shouldn’t use floats as Boolean implicitly. Do you mind changing this to “if (restvalue != 0.0f)” and the same below and doing a force push? I had misread it before as a bool and it would be much clearer. (And since you are there, this looks 4 tabbed; did you run clang-format vs master? No big deal there tho).

Otherwise all LGTM! If you force push that change I’m happy to merge it.